server/start.go

```diff
@@ -107,26 +108,64 @@ func startInProcess(svrCtx *server.Context, svrCfg serverconfig.Config, clientCt
 ) error {
 	cmtCfg := svrCtx.Config
 	gRPCOnly := svrCtx.Viper.GetBool(flagGRPCOnly)
-	g, ctx := getCtx(svrCtx, true)
+	g, ctx := getCtx(svrCtx, true) // Get errgroup and context
+
+	var rpcServer *rpc.RPCServer // Declare rpcServer variable
 
 	if gRPCOnly {
 		// TODO: Generalize logic so that gRPC only is really in startStandAlone
 		svrCtx.Logger.Info("starting node in gRPC only mode; CometBFT is disabled")
 		svrCfg.GRPC.Enable = true
 	} else {
 		svrCtx.Logger.Info("starting node with ABCI CometBFT in-process")
-		_, rpcServer, cleanupFn, err := startNode(ctx, svrCtx, cmtCfg, app)
+		// Call startNode to initialize components
+		rollkitNode, localRpcServer, executor, cleanupFn, err := startNode(ctx, svrCtx, cmtCfg, app)
 		if err != nil {
 			return err
 		}
 		defer cleanupFn()
+		rpcServer = localRpcServer
+
+		svrCtx.Logger.Info("Attempting to start RPC server")
+		if err := rpcServer.Start(); err != nil {
+			svrCtx.Logger.Error("Failed to start RPC server", "error", err)
+			return fmt.Errorf("failed to start abci rpc server: %w", err)
+		}
+		svrCtx.Logger.Info("RPC server started successfully")
+
+		g.Go(func() error {
+			svrCtx.Logger.Info("Attempting to start Rollkit node run loop")
+			err := rollkitNode.Run(ctx)
+			if err != nil && err != context.Canceled {
+				svrCtx.Logger.Error("Rollkit node run loop failed", "error", err)
+				return fmt.Errorf("rollkit node run failed: %w", err)
+			}
+			if err == context.Canceled {
+				svrCtx.Logger.Info("Rollkit node run loop cancelled by context")
+			} else {
+				svrCtx.Logger.Info("Rollkit node run loop completed")
+			}
+			return nil // Return nil on graceful shutdown or normal completion
+		})
+		svrCtx.Logger.Info("Rollkit node run loop launched in background goroutine")
+
+		// Wait for the node to start p2p before attempting to start the gossiper
+		time.Sleep(1 * time.Second)
```

Comment:
Good catch, in the future we could add a method to check if P2P is running or something like that